### PR TITLE
add: new token balance view

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
this moves tokenbalance from side-bar to top-nav

- [x] - add balance + icon to topnav for trigger the balance view
- [x] - show token-balance in the window opened
- [x] - add close when clicking on the bottom of the window 

About # (link your issue here)
#498 

# Screens
(the double bars at the bottom of the balanceview window are already added, just not on the screens shown)
image

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/components/WalletBalance/index.tsx [hunk 1]
```diff
@@ -13,7 +27,140 @@ export type WalletBalanceProps = {
...
+        })
+        allTokens.push({
+            type: 'ERC20',
+            options: {
+                address: GOOD.address,
+                symbol: GOOD.ticker,
+                decimals: GOOD.decimals,
+                image: 'https://raw.githubusercontent.com/GoodDollar/GoodProtocolUI/master/src/assets/images/tokens/good-logo.png',
+            },
+        })
```
Comment (johnsmith-gooddollar) [inline src/components/WalletBalance/index.tsx:60]:
```suggestion
        const allTokens: any[] = [
        {
            type: 'ERC20',
            options: {
                address: G$.address,
                symbol: G$.ticker,
                decimals: G$.decimals,
                image: 'https://raw.githubusercontent.com/GoodDollar/GoodProtocolUI/master/src/assets/images/tokens/gd-logo.png',
            },
        }, {
            type: 'ERC20',
            options: {
                address: GOOD.address,
                symbol: GOOD.ticker,
                decimals: GOOD.decimals,
                image: 'https://raw.githubusercontent.com/GoodDollar/GoodProtocolUI/master/src/assets/images/tokens/good-logo.png',
            },
        }]
```

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: src/components/AppBar.tsx [hunk 2]
```diff
@@ -205,13 +209,16 @@ const Web3Bar = () => {
...
-    const { chainId } = useActiveWeb3React()
+    const { account, chainId } = useActiveWeb3React()
     const [sidebarOpen, setSidebarOpen] = useState(false)
     const isSimpleApp = useIsSimpleApp()
     const showPrice = useFeatureFlagEnabled('show-gd-price')
     const posthog = usePostHog()
     const payload = posthog?.getFeatureFlagPayload('app-notice')
     const { enabled: appNoticeEnabled, message, color, link } = (payload as any) || {}
+    const [walletBalanceOpen, setWalletBalanceOpen] = useState(false)
+    const { G$ } = useG$Balance(5)
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply src/components/AppBar.tsx:220]: this is refresh rate, not chainId

File: src/components/AppBar.tsx [hunk 3]
```diff
@@ -205,13 +209,16 @@ const Web3Bar = () => {
...
+    const [gdBalance, setGdBalance] = useState('0.00')
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply src/components/AppBar.tsx]: done

File: src/components/WalletBalance/index.tsx (hunk 1 above)
Comment (L03TJ3) [reply src/components/WalletBalance/index.tsx]: done

File: src/components/AppBar.tsx [hunk 4]
```diff
@@ -222,6 +229,17 @@ function AppBar(): JSX.Element {
...
+        if (G$) {
+            const formatted = G$.format({
+                useFixedPrecision: true,
+                suffix: '',
+                prefix: G$.currency?.ticker + ' ',
+            })
+            setGdBalance(formatted)
+        }
+    }, [/* used */ chainId, G$])
+
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply src/components/AppBar.tsx]: done

**Review by johnsmith-gooddollar — APPROVED** [review]

**Pull request merged**